tests/vm_test.go

```diff
@@ -980,12 +981,12 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 					Eventually(ThisVM(vm), 240*time.Second, 1*time.Second).Should(Not(haveStateChangeRequests()))
 				})
 
-				It("[test_id:2188] should not remove a succeeded VMI", func() {
+				It("[test_id:2188] should remove a succeeded VMI", func() {
```

Comment:
Why do we change this?

Reply:
The hack of keeping the succeeded VMI around as a boolean information is not longer necessary for the RerunOnFailure strategy to function properly.
This was also discussed there: https://github.com/kubevirt/kubevirt/pull/10639#discussion_r1423168215

Reply:
While we could consider it as a hack I still think it is good to keep around. Consider a case when we create a VM, it succeeded right away and we don't see anything anymore. (IMO this is the same behavior as for Pods with other controllers.)
Can we keep the behavior?

Reply:
Succeeded VMIs are already removed right away with every other run strategy.
We kept it around only for RerunOnFailure to know that last VMI succeeded and it should not start the VM again.
If we want to keep successful VMIs around, we should do it as a separate effort and with all run strategies IMO.

Reply:
Right that is correct because we can keep the VMI only in case of `RerunOnFailure` and `Once`